java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/internal/Expectations.java

```diff
@@ -180,17 +189,77 @@ public String toString() {
     }
   }
 
+  private static class EditComment {
+    final Map<IssueAttribute, Object> attributes;
+
+    public EditComment(Issue issue) {
+      attributes = issue;
+    }
+
+    String replacement() {
+      String replacement = (String) attributes.get(MESSAGE);
+      if (replacement == null) {
+        throw new AssertionError("Quickfix edit should contain a replacement.");
+      }
+      return replacement;
+    }
+
+    AnalyzerMessage.TextSpan getTextSpan(int line) {
+      // Line of the actual comment, used for reporting missing elements.
+      Integer editLine = (Integer) attributes.get(LINE);
```

Comment:
Integer editLine = LINE.get(attributes);

      int startLine = getAbsoluteLine(START_LINE.get(attributes), line);
      int startColumn = getColumnOffset(START_COLUMN.get(attributes), "start", editLine);
      int endColumn = getColumnOffset(END_COLUMN.get(attributes), "end", editLine);
      Integer endLineAttribute = END_LINE.get(attributes);

Reply:
Note that we can not do this for lines, because we want to call `getLine()` with line reference "line".
It works fine for all others cases though.